### PR TITLE
Add support for growth mode, aka log-log plot of recent changes

### PR DESCRIPTION
**Pull request by davidebbo**
This is the initial change for #12. It turned out to be relatively straightforward.

Note that I tried the match the behavior of https://aatishb.com/covidtrends/, which is what that youtube video is showing. The resulting graph closely matches that site, so there is some confidence that it's correct. e.g. try testing it with `?include=China%3BAustralia%3BJapan%3BS%20Korea&advanced=1&start=>%3D30&domain=Intl&stat=growth`.

image

Misc comments:
- I called it 'growth' for lack of a better short name
- The graph looks a bit rough with low numbers when using log scale (long straight initial line). It's best to start with >= 80
- It can be used in either log/log mode or linear/linear mode. Obviously, log is more interesting.
- The axis names are wrong, as I have not touch that logic and I need to customize it for this new mode. I'll look at that.
- In log mode, the x axis labels are missing. Not sure why, but I'll look into that.

### CODE REVIEW

**Comment by davidebbo** [comment]
I fixed the axis titles.

But I also found a bug: when setting the start to `>=10', it blows up in chartist logic. I need to figure out why...

**Comment by davidebbo** [comment]
Ok, fixed both that crash and the missing x axis labels.

File: index.html [hunk 1]
```diff
@@ -570,9 +575,23 @@ <h1>{{ chosen_domain == 'US' ? 'USA' :
...
+      delete point["value"]; // delete unwanted 'value' prop as we have x,y
+      return point;
+    })]);
+  }
   if (scale.startsWith('log')) {
+    // When using log scale, we need to make sure we don't end up with entries
+    // that are missing their 'value' (or 'x/y') properties, as that breaks it
     extract.series = _.map(extract.series, p => [p[0], _.map(p[1], d => {
-      return d && d.value && d.value > 0 ? d : null;
+      return d && (d.value || (d.x && d.y)) ? d : null;
```
Comment (davidebbo) [inline index.html:594]: Note: I removed the `d.value > 0` check which didn't seem necessary.No

File: index.html [hunk 2]
```diff
@@ -559,6 +563,7 @@ <h1>{{ chosen_domain == 'US' ? 'USA' :
      select, updateSelect) {
   scale = scale || 'log10';
   if (chart) { chart.detach(); }
+  var isGrowthChart = (stat == 'growth');
```
Comment (davidebbo) [inline index.html:566]: I tried to reduce the number of places we test for the literal string, but still ended up with a few. Ideally, we'd have a more formal data model to avoid dealing with string settings all the way through.

File: index.html (hunk 1 above)
Comment (davidbau) [reply index.html:594]: The d.value > 0 was to avoid negative numbers that sometimes showed up in daily deltas, messing up log plots.  I'll go test it now.

Comment (davidbau) [reply index.html:594]: Yeah - missing the value >0 check causes a crash on this URL:

http://localhost:8000/#/?start=2%2F29%2F20&series=deaths&stat=daily&include=NY&top=0

Let's add the d.value > 0 check back (and to match you might want to verify x and y > 0 also even though the data source probably has no negatives right now).

File: index.html (hunk 2 above)
Comment (davidbau) [reply index.html:566]: OK.  I think this hack is clean enough for now.

File: index.html [hunk 3]
```diff
@@ -538,7 +539,10 @@ <h1>{{ chosen_domain == 'US' ? 'USA' :
              (scale && scale.startsWith('log') ? ' (log scale)' : '');
   return desc;
 }
-function xaxis_title(field, start, norm) {
+function xaxis_title(field, stat, scale, norm, start) {
+  if (stat == 'growth') {
+    return yaxis_title(field, 'totals', scale, norm);
```
Comment (davidbau) [inline index.html:544]: Ah clever, reuse yaxis_title ha!

File: index.html (hunk 1 above)
Comment (davidebbo) [reply index.html]: Good catch, I brought it back and added it for x & y
